> Can you please explain more of how this querying of the parent card is useful? What is a typical use case here?
> 
> Also, my CI is reporting the following type errors:
> 
> ```
> jfa/list_issues.py:65: error: "object" has no attribute "get"  [attr-defined]
> jfa/list_issues.py:66: error: "object" has no attribute "get"  [attr-defined]
> jfa/list_issues.py:77: error: Missing key "issue_status" for TypedDict "ResultVariables"  [typeddict-item]
> jfa/list_issues.py:77: error: Extra keys ("parent_key", "parent_summary") for TypedDict "ResultVariables"  [typeddict-unknown-key]
> ```

Hey @caleb531 

I wasn't aware of how `TypedDict` worked - so I *think* I updated all that (hopefully) correctly - curious to see if we get a green build 🤞

I reverted all the dictionary safety since it shouldn't be needed since we are explicitly setting the types

For your first question - I updated my local version of this workflow to show the parent (aka an Epic) and I thought it was really helpful. I am a tech lead for a software company and we deal a lot with Epics. Every card needs to have an epic, etc. And when I am moving cards around or creating cards, or just looking at old cards, it is nice to know which Epic it is at a glance

If it adds too much bloat - we don't have to merge this in - but wanted to share just incase!

--- a/jfa/list_issues.py
+++ b/jfa/list_issues.py
@@ -56,16 +56,20 @@ def get_issue_url(issue_key: str) -> str:
 # Convert the given dictionary representation of a Jira issue to a Alfred
 # feedback result dictionary
 def get_result_from_issue(issue: Issue) -> Result:
-    fields = issue.get("fields", {})
+    fields = issue["fields"]
 
-    issue_type = fields.get("issuetype", {}).get("name", "").lower()
-    issue_key = issue.get("key", "")
-    issue_summary = fields.get("summary", "")
-    issue_status = fields.get("status", {}).get("name", "")
+    issue_type = fields["issuetype"]["name"].lower()
+    issue_key = issue["key"]
+    issue_summary = fields["summary"]
+    issue_status = fields["status"]["name"]
 
     parent = fields.get("parent")
-    parent_key = parent.get("fields", {}).get("key") if parent else None
-    parent_summary = parent.get("fields", {}).get("summary") if parent else None
+    parent_key = None
+    parent_summary = None
+    if parent:
+        parent_fields = parent["fields"]
+        parent_key = parent_fields.get("key")
+        parent_summary = parent_fields.get("summary")
 
     subtitle = f"{issue_key} ({issue_status})"
     if parent_summary:
@@ -74,10 +78,10 @@ def get_result_from_issue(issue: Issue) -> Result:
     return {
         "title": issue_summary,
         "subtitle": subtitle,
-        "arg": issue.get("id", ""),
+        "arg": issue["id"],
         "icon": {"path": get_issue_type_icon(issue)},
         "variables": {
-            "issue_id": issue.get("id", ""),
+            "issue_id": issue["id"],
             "issue_key": issue_key,
             "issue_type": issue_type,
             "issue_summary": issue_summary,

--- a/jfa/types.py
+++ b/jfa/types.py
@@ -2,6 +2,14 @@
 
 from typing import Optional, TypedDict
 
+class ParentFields(TypedDict):
+    key: str
+    summary: str
+
+
+class Parent(TypedDict):
+    fields: ParentFields
+
 
 class IssueType(TypedDict):
     name: str
@@ -17,6 +25,7 @@ class IssueFields(TypedDict):
     summary: str
     issuetype: IssueType
     status: IssueStatus
+    parent: Optional[Parent]
 
 
 class Issue(TypedDict):
@@ -36,6 +45,8 @@ class ResultVariables(TypedDict):
     issue_summary: str
     issue_url: str
     issue_status: str
+    parent_key: Optional[str]
+    parent_summary: Optional[str]
 
 
 class ResultText(TypedDict, total=False):

--- a/jfa/types.py
+++ b/jfa/types.py
@@ -2,6 +2,7 @@
 
 from typing import Optional, TypedDict
 
+
 class ParentFields(TypedDict):
     key: str
     summary: str

